acasclient/lsthing.py

```diff
@@ -2076,6 +2109,62 @@ def _upload_file_values_from_state_dict(state_dict):
             return state_dict
         self.metadata = _upload_file_values_from_state_dict(self.metadata)
         self.results = _upload_file_values_from_state_dict(self.results)
+    
+    def _get_ddicts(self):
+        """Extract a unique list of DDict objects from the CodeValue attributes of a SimpleThing
+        :return: dictionary of { (code_type, code_kind, code_origin): DDict }
+        """
+        ddicts = {}
+        state_dicts = [self.metadata, self.results]
+        for state_dict in state_dicts:
+            for _, values_dict in state_dict.items():
```

Comment:
You could alternatively do:
```python
for values_dict in state_dict.values():
    ...
```

It could be the same with for-loops where we are ignoring the key.